---
title: "Background knowledge for the Tidy Transcriptomics workshop"
author:
  - Maria Doyle, Peter MacCallum Cancer Centre^[maria.doyle@petermac.org]
  - Stefano Mangiola, Walter and Eliza Hall Institute^[mangiola.s@wehi.edu.au]
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Background knowledge for the Tidy Transcriptomics workshop}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```


```{r, echo=FALSE, out.width = "100px"}
knitr::include_graphics("../inst/vignettes/tidybulk_logo.png")
```

# Schedule

Schedule can be found [here](https://tidy-transcriptomics-workshops.github.io/ismb2021_tidytranscriptomics/index.html#time-outline).  
<br>
**Format**: Hands on demos, exercises plus Q&A 
**Interact**: Zoom chat
<br>
<br>


# What is transcriptomics?

“The transcriptome is the set of all RNA transcripts, including coding and non-coding, in an individual or a population of cells”

_Wikipedia_

```{r, echo=FALSE, out.width = "600px"}
knitr::include_graphics("../inst/vignettes/transcriptomics.jpg")
```

# Why use transcriptomics?

- Genome (DNA) pretty stable
- Proteome (proteins) harder to measure
- Transcriptome (RNA) can measure changes in expression of thousands of coding and non-coding transcripts
<br>
<br>

# Possible experimental design

```{r, echo=FALSE, out.width = "600px"}
knitr::include_graphics("../inst/vignettes/ScreenShot2.png")
```

# How does transcriptomics work?

```{r, echo=FALSE, out.width = "600px"}
knitr::include_graphics("../inst/vignettes/ScreenShot3.png")
```

# Types of transcriptomic analyses

* **Differential expression**  
* **Cell type composition**  
* Alternative splicing  
* Novel transcript discovery  
* Fusions identification  
* Variant analysis  
<br>
Topics in bold we will see in this workshop
<br>
<br>

# Bulk RNA sequencing differential expression workflow

```{r, echo=FALSE, out.width = "600px"}
knitr::include_graphics("../inst/vignettes/bulk_RNAseq_pipeline.png")
```

# Differences between bulk and single-cell RNA sequencing

```{r, echo=FALSE, out.width = "600px"}
knitr::include_graphics("../inst/vignettes/bulk_vs_single.jpg")
```

_Shalek and Benson, 2017_

# Single-cell RNA sequencing analysis workflow

```{r, echo=FALSE, out.width = "600px"}
knitr::include_graphics("../inst/vignettes/single_cell_RNAseq_pipeline.png")
```

# Tidy data and the tidyverse

This workshop demonstrates how to perform analysis of RNA sequencing data following the tidy data paradigm [@wickham2014tidy]. The tidy data paradigm provides a standard way to organise data values within a dataset, where each variable is a column, each observation is a row, and data is manipulated using an easy-to-understand vocabulary. Most importantly, the data structure remains consistent across manipulation and analysis functions. For more information, see the R for Data Science chapter on tidy data [here](https://r4ds.had.co.nz/tidy-data.html).

```{r, echo=FALSE, out.width = "500px", fig.cap="[Tidy data for efficiency, reproducibilty and collaboration](https://www.openscapes.org/blog/2020/10/12/tidy-data/)"}
knitr::include_graphics("../inst/vignettes/tidydata_1.jpg")
```

The tidyverse is a collection of packages that can be used to tidy, manipulate and visualise data. We'll use many functions from the tidyverse in this workshop, such as `filter`, `select`, `mutate`, `pivot_longer` and `ggplot`.

```{r, echo=FALSE, out.width = "500px", fig.cap="Adapted from [Getting Started with tidyverse in R](https://www.storybench.org/getting-started-with-tidyverse-in-r/)"}
knitr::include_graphics("../inst/vignettes/tidyverse.png")
```

# Getting started 

## Cloud

Easiest way to run this material. Only available during workshop. Many thanks to the Australian Research Data Commons (ARDC) for providing RStudio in the Australian Nectar Research Cloud and Andy Botting from ARDC for helping to set up.

 * Login to one of the servers (e.g. https://rstudio1.tidytranscriptomics-workshop.cloud.edu.au/rstudio/auth-sign-in) with one of the supplied usernames and passwords
 * Open `tidytranscriptomics.Rmd` in `vignettes` folder
 

## Local

We will use the Cloud during the ISMBECCB workshop and this method is available if you want to run the material after the workshop. If you want to install on your own computer, see instructions [here](https://tidy-transcriptomics-workshops.github.io/ismb2021_tidytranscriptomics/index.html#workshop-package-installation). 

Alternatively, you can view the material at the workshop webpage [here](https://tidy-transcriptomics-workshops.github.io/ismb2021_tidytranscriptomics/articles/tidytranscriptomics.html). 
